### PR TITLE
Add actionless flag and add tests

### PR DESCRIPTION
**Pull request by roeiK-wix**
Add actionless flag, very similar to dry flag but the only difference is that all the probes will be executed but the activities of the actions will not be executed.

### CODE REVIEW

**Comment by alexander-gorelik** [comment]
Comes in tandem with https://github.com/chaostoolkit/chaostoolkit/pull/220

**Comment by alexander-gorelik** [comment]
@Lawouach 
Super useful when you develop an experiment

- To run and validate all no destructive parts
- To run them. fast without waiting for all the defined pauses
- Before that flag, we were forced to create an additional experiment file with no pauses and no actions.

Super other useful things

- We are planning to run experiments nightly with this flag to see that all the moving parts and external system dependencies of the experiment like monitoring, logging, except the destructive actions themselves are working properly
- The nightly run with the flag also validates the integrity and workability of experiments' common code that contributed by different engineers from different teams in addition to unit tests without any production risk.

**Comment by Lawouach** [comment]
I like the idea and thank you for providing context.

Before I can fully review it, one quick immediate note. Let's change the name to a more consistent one. 

```
--dry => nothing is actually executed
--dry-actions => only probes are executed
--dry-probes => only actions are executed
```

Or, more sophisticated, but likely not really needed:

```
--dry
--dry-strategy=actions|probes
```

for instance:

```
$ chaos run --dry --dry-strategy=actions ...
```

would mean, do not run any actions

This offers greater flexibility for the future but adds complexity.

**Comment by roeiK-wix** [comment]
I understand what you are saying, but we are not seeing any use-case of using dry-probes, because usually, the probes are not dangerous, and moreover when you have probes and actions you don't want to run only actions without probes.
If you have some use-case that you think of, please elaborate.

**Comment by roeiK-wix** [comment]
Is there any progress on that?

**Comment by Lawouach** [comment]
Hello @roeiK-wix 

I was on vacation so no. No updates and no ETA.

I'm not convinced by the naming of the flags still. Bit I like the feature.
My concern is about ensuring consistency in flags/features of the CTK.

**Comment by alexander-gorelik** [comment]
@Lawouach we want to implement that and we have time, it is just hanging.
```--dry-actions``` it is good enough, in my opinion, shall we proceed with that implementation?

**Comment by Lawouach** [comment]
Sounds good.

Would you be kind to at least look at the flag being `--dry=actions` please? This would allow for `--dry=controls` for example too and reduce the number of new flags.

Cheers,

**Comment by roeiK-wix** [comment]
Hey, we finished refactoring the dry flag according to what we agreed on. 
Flag options:
 `--dry=actions` - Not executing any actions in the method and rollback sections.
`--dry=probes`  - Not executing any probes in the method and rollback sections.
`--dry=activityies`  - Not executing any actions and probes in the method and rollback sections. (the same as the old dry flag)
`--dry=pause`  - Not executing any pause. 
During the implementation, we found that the old and now the new dry flag are just working inside the method section.
But in all the other sections such as the steady-state and in the controls, the activities are still working as normal (meaning no dry)
We wanted to know if this is the logic you meant to when you thought about adding a dry flag?
Also, we found that the probes aren't returning any value, (as expected to a function that doesn't been called) so it always returns None and so it's always false tolerance.

In this new implementation, we kept the old logic and added the new logic with actionless, probeless, and pauseless.

Also, review this PR -> https://github.com/chaostoolkit/chaostoolkit/pull/220

We didn't add any tests for this new logic because we wanted you to first review this PR and then we will add the tests.
Thanks.
@Lawouach

File: chaoslib/hypothesis.py [hunk 1]
```diff
@@ -203,7 +203,7 @@ def run_steady_state_hypothesis(experiment: Experiment,
 
             run["tolerance_met"] = True
 
-            if dry:
+            if dry == "probes" or dry == "activities":
```
Comment (Lawouach) [inline chaoslib/hypothesis.py:206]:
```suggestion
            if dry in ("probes", "activities"):
```

File: chaoslib/rollback.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
 
 def run_rollbacks(experiment: Experiment, configuration: Configuration,
                   secrets: Secrets, pool: ThreadPoolExecutor,
-                  dry: bool = False) -> Iterator[Run]:
+                  dry: str = "no-dry") -> Iterator[Run]:
```
Comment (Lawouach) [inline chaoslib/rollback.py:16]: same

File: chaoslib/run.py [hunk 3]
```diff
@@ -331,10 +331,19 @@ def _run(self, strategy: Strategy, schedule: Schedule,  # noqa: C901
         hypo_pool = get_hypothesis_pool()
         continous_hypo_event = threading.Event()
 
-        dry = experiment.get("dry", False)
-        if dry:
+        dry = experiment.get("dry", "no-dry")
```
Comment (Lawouach) [inline chaoslib/run.py:334]:
```suggestion
        dry = experiment.get("dry", None)
```

to reflect the switch from `"no-dry"` to `None`

File: chaoslib/run.py [hunk 4]
```diff
@@ -458,7 +467,7 @@ def should_run_during_method(strategy: Strategy) -> bool:
 def run_gate_hypothesis(experiment: Experiment, journal: Journal,
                         configuration: Configuration, secrets: Secrets,
                         event_registry: EventHandlerRegistry,
-                        dry: bool = False) -> Dict[str, Any]:
+                        dry: str = "no-dry") -> Dict[str, Any]:
```
Comment (Lawouach) [inline chaoslib/run.py:470]: same

File: chaoslib/run.py [hunk 5]
```diff
@@ -487,7 +496,7 @@ def run_deviation_validation_hypothesis(experiment: Experiment,
                                         configuration: Configuration,
                                         secrets: Secrets,
                                         event_registry: EventHandlerRegistry,
-                                        dry: bool = False) \
+                                        dry: str = "no-dry") \
```
Comment (Lawouach) [inline chaoslib/run.py:499]: same

**Review by Lawouach — COMMENTED** [review]
LGTM in principles.

Please look at my changes proposal.

Also, please make sure to thoroughly document the change in the CHANGELOG please.

**Comment by Lawouach** [comment]
Also there is a conflict, so you may have to rebase from master first.

File: chaoslib/activity.py [hunk 6]
```diff
@@ -147,11 +147,22 @@ def execute_activity(experiment: Experiment, activity: Activity,
...
+        elif dry == "activities":
+            is_dry = True
+
         if pause_before:
             logger.info("Pausing before next activity for {d}s...".format(
                 d=pause_before))
-            # only pause when not in dry-mode
-            if not dry:
+            # pause when one of the dry flags are set
+            if dry != "pause" and not is_dry:
```
[the remark this answers is not in the document]
Comment (roeiK-wix) [reply chaoslib/activity.py:165]: Yep, it will be a nice feature.

**Comment by Lawouach** [comment]
Hey folks,

It seems GitHub had a glitch. Your forked branch didn't seem to have any commit so when we merged another of our branches, GH seems to have considered this meant this one could be marked as merged too.

Feel free to continue using it.

**Pull request merged**